lib/tasks/user_mapping.rake

```diff
@@ -0,0 +1,77 @@
+require 'json'
+require 'aws-sdk'
+
+namespace :user_mapping do
+  task :configure => :environment do |t, args|
+    Aws.config.update(
+      endpoint: Rails.configuration.x.backup_s3_endpoint,
+      access_key_id: Rails.configuration.x.backup_s3_access_key_id,
+      secret_access_key: Rails.configuration.x.backup_s3_secret_access_key,
+      force_path_style: true,
+      region: Rails.configuration.x.backup_aws_region,
+    )
+  end
+
+  task :create_bucket => [:environment, :configure] do |t, args|
+    Aws::S3::Client.new.create_bucket({
+      bucket: Rails.configuration.x.backup_s3_bucket_name
+    })
+  end
+
+  task :push => :configure do |t, args|
+
+    # https://docs.aws.amazon.com/sdk-for-ruby/v2/api/Aws/S3/Client.html
+    s3_client = Aws::S3::Client.new
+
+    # https://docs.aws.amazon.com/sdk-for-ruby/v2/api/Aws/S3/Resource.html
+    bucket = Aws::S3::Resource.new(client: s3_client).bucket(Rails.configuration.x.backup_s3_bucket_name)
+
+    # SQL query to get the data we need:
+    # select
+    #   u.id,
+    #   s.user_name as slack_nickname,
+    #   g.login github_login,
+    #   st.domain as slack_domain,
+    #   st.github_organization,
+    # from users u
+    # inner join slack_accounts s on s.user_id = u.id
+    # inner join github_accounts g on g.user_id = u.id
+    # inner join slack_teams st on s.slack_team_id = st.id;
+    #
+    # Go through each Slack team with have in the DB.
+    SlackTeam.all.each do |team|
+      # Get all the users for the team. We only care about the Slack <-> GitHub
+      # mapping.
+      query = User.select(
+        :id,
+        "slack_accounts.user_name as slack_nickname",
+        "github_accounts.login as github_login",
+      ).joins(
+        :slack_accounts,
+        :github_accounts
+      ).distinct
```

Comment:
I believe this should be relatively efficient query, doing the actual joins in SQL rather than activerecord, but it doesn't look like like the foreign keys are indexed, so performance would be poor with a large dataset. That's a separate issue from this PR though, and in reality the dataset is so small it won't make any difference.

Reply:
Running an explain shows a sequential scan for the joins.